### PR TITLE
feat(scaffolder): add a `publish:gerrit:review` action

### PR DESCRIPTION
**Pull request by tcardonne**
## Hey, I just made a Pull Request!

This PR adds a `publish:gerrit:review` Scaffolder action that creates a Gerrit review.
Gerrit reviews can be created by pushing a commit on `refs/for/` namespace (see https://gerrit-review.googlesource.com/Documentation/concept-refs-for-namespace.html).

To achieve this we also had to change the `isomorphic-git` wrapper to allow specifying the remoteRef and add some helper functions.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] ~Added or updated documentation~
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] ~Screenshots attached (for UI changes)~
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gerritReview.test.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        ...mockContext,
+        input: {
+          repoUrl: 'missing.com?workspace=w&owner=o&repo=repo',
+        },
+      }),
+    ).rejects.toThrow(/No matching integration configuration/);
+  });
+
+  it('can correctly create a review', async () => {
+    // expect.assertions(5);
```
[the remark this answers is not in the document]
Comment (tcardonne) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gerritReview.test.ts]: Woops, commented it while iterating, forgot to uncomment it with the correct expected value.

**Comment by tcardonne** [comment]
No worries, I've added a second changeset for the `@backstage/backend-common` package.

**Comment by tcardonne** [comment]
No worries, I've added a second changeset for the `@backstage/backend-common` package.

**Review by benjdlambert — APPROVED** [review]
Cool! Thanks so much :pray:

**Pull request merged**